### PR TITLE
Update AzureML step operator to SDK v2 and add service connector support

### PR DESCRIPTION
**Pull request by stefannica**
## Describe changes
Makes a few improvements to the AzureML step operator and orchestrator:

* migrates the AzureML step operator code to use AzureML Python SDK v2. This aligns nicely with the code used by the recently introduced AzureML orchestrator. This also changes how the step operator behaves: before, the container image build was performed on-the-fly, as part of the step operator run; now, the step operator needs a container registry and an image is built in advance, just as is usual with other containerized step operators and orchestrators. An AzureML compute is also no longer required to be provisioned and configured ahead of time and the AzureML step operator defaults to using the serverless compute option.
* adds service connector support to the AzureML step operator
* adds synchronous support to the AzureML orchestrator

Most of these changes are required to be able to offer a similar stack deployment experience via one-click, wizard and Terraform.

## Breaking Changes

The AzureML step operator now requires a container registry to be part of the same stack. An AzureML compute is also no longer required for the step operator to function. It will use serverless compute by default.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [x] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Review by avishniakov — APPROVED** [review]

File: src/zenml/integrations/azure/step_operators/azureml_step_operator.py [hunk 1]
```diff
@@ -13,50 +13,35 @@
...
 
-import zenml
-from zenml.client import Client
-from zenml.constants import (
-    ENV_ZENML_CONFIG_PATH,
-)
-from zenml.environment import Environment as ZenMLEnvironment
+from zenml.config.build_configuration import BuildConfiguration
+from zenml.enums import StackComponentType
+from zenml.integrations.azure.azureml_utils import create_or_get_compute
```
Comment (schustmi) [inline src/zenml/integrations/azure/step_operators/azureml_step_operator.py:25]: Maybe i'm missing something, but both this file and wherever `AzureMLComputeSettings` are defined are missing?

**Review by schustmi — CHANGES_REQUESTED** [review]

File: src/zenml/integrations/azure/step_operators/azureml_step_operator.py (hunk 1 above)
Comment (bcdurak) [reply src/zenml/integrations/azure/step_operators/azureml_step_operator.py:25]: I was thinking the exact same thing 😄

Comment (stefannica) [reply src/zenml/integrations/azure/step_operators/azureml_step_operator.py:25]: yes, true, I just added them

**Review by schustmi — APPROVED** [review]
Finally the azureml step operator is fully supporting our docker settings 🙏

File: docs/book/component-guide/step-operators/azureml.md [hunk 2]
```diff
@@ -15,10 +15,17 @@ You should use the AzureML step operator if:
 
 ### How to deploy it
 
-* Create a `Machine learning` [resource on Azure](https://docs.microsoft.com/en-us/azure/machine-learning/quickstart-create-resources) .
-* Once your resource is created, you can head over to the `Azure Machine Learning Studio` and [create a compute cluster](https://docs.microsoft.com/en-us/azure/machine-learning/quickstart-create-resources#cluster) to run your pipelines.
-* Create an `environment` for your pipelines. Follow [this guide](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-manage-environments-in-studio) to set one up.
-* (Optional) Create a [Service Principal](https://docs.microsoft.com/en-us/azure/developer/java/sdk/identity-service-principal-auth) for authentication. This is required if you intend to run your pipelines with a remote orchestrator.
+{% hint style="info" %}
+Would you like to skip ahead and deploy a full ZenML cloud stack already,
+including an AzureML step operator? Check out the [in-browser stack deployment wizard](../../how-to/stack-deployment/deploy-a-cloud-stack.md),
```
Comment (bcdurak) [inline docs/book/component-guide/step-operators/azureml.md:20]: Is this possible? AFAIK, step operators are not supported by the 1-click deployment or the stack wizard, right?

**Review by bcdurak — APPROVED** [review]
Aside from a small comment regarding the docs, LGTM.

File: docs/book/component-guide/step-operators/azureml.md (hunk 2 above)
Comment (stefannica) [reply docs/book/component-guide/step-operators/azureml.md:20]: they will be starting with the next release :) I added step operators to all of our terraform modules

**Pull request merged**